canonical/maas-ui
# feat(node): Show IP address in node summary network card (#2851)

## Done

- Added interface IP address to node summary network card
- Moved MAC address beneath Name
- Added subnet beneath IP address
- Added vlan beneath Fabric

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Navigate to the summary page for a device
- Ensure the network card shows IP addresses for interfaces in the right column (may show "DHCP" or "Unconfigured")
- Ensure a subnet is shown beneath an IP address (if one is set for that interface)
- Ensure the MAC address is shown beneath interface names
- Ensure the VLAN (if set) is shown beneath fabrics

## Fixes

Fixes #2851 

## Launchpad issue

[lp#1881560](https://bugs.launchpad.net/maas/+bug/1881560)

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots

### Before 
![image](https://user-images.githubusercontent.com/35104482/216576363-3e742044-2188-489b-bbab-fdc497b814bb.png)

### After
![image](https://user-images.githubusercontent.com/35104482/216576277-1262c68a-a49e-434d-921d-a64851074956.png)
